src/StatsdClient/DogStatsdService.cs

```diff
@@ -242,6 +242,15 @@ public void ServiceCheck(string name, Status status, int? timestamp = null, stri
             _metricsSender?.SendServiceCheck(name, (int)status, timestamp, hostname, tags, message);
         }
 
+        /// <summary>
+        /// Flushes all metrics.
+        /// </summary>
+        public void Flush()
+        {
+            _statsdData?.Flush();
+            _statsdData = null;
```

Comment:
you dont need to create a new `_statsdData`?

Reply:
Fixed.